joint_state_controller:
  type:         joint_state_controller/JointStateController
  publish_rate: 60

joint_trajectory_controller: 
  type: effort_controllers/JointTrajectoryController  
  joints:
    - ur5_shoulder_pan_joint
    - ur5_shoulder_lift_joint
    - ur5_elbow_joint
    - ur5_wrist_1_joint
    - ur5_wrist_2_joint
    - ur5_wrist_3_joint
  gains:
    ur5_shoulder_pan_joint: {p: 2000, i: 100, d: 50}
    ur5_shoulder_lift_joint: {p: 2000, i: 100, d: 50}
    ur5_elbow_joint: {p: 2000, i: 100, d: 50}
    ur5_wrist_1_joint: {p: 2000, i: 100, d: 50}
    ur5_wrist_2_joint: {p: 2000, i: 100, d: 50}
    ur5_wrist_3_joint: {p: 2000, i: 100, d: 50}
  constraints:
    goal_time: 0.5
    ur5_shoulder_pan_joint: { goal: 0.05 }
    ur5_shoulder_lift_joint: { goal: 0.05 }
    ur5_elbow_joint: { goal: 0.05 }
    ur5_wrist_1_joint: { goal: 0.05 }
    ur5_wrist_2_joint: { goal: 0.05 }
    ur5_wrist_3_joint: { goal: 0.05 }

wheel_back_left_joint_velocity_controller:
  type:         effort_controllers/JointVelocityController
  joint:        wheel_back_left_joint
  pid:          {p: 100, i: 1, d: 0.1}

wheel_back_right_joint_velocity_controller:
  type:         effort_controllers/JointVelocityController
  joint:        wheel_back_right_joint
  pid:          {p: 100, i: 1, d: 0.1}

wheel_front_left_joint_velocity_controller:
  type:         effort_controllers/JointVelocityController
  joint:        wheel_front_left_joint
  pid:          {p: 100, i: 1, d: 0.1}

wheel_front_right_joint_velocity_controller:
  type:         effort_controllers/JointVelocityController
  joint:        wheel_front_right_joint
  pid:          {p: 100, i: 1, d: 0.1}

ur5_shoulder_pan_joint_position_controller:
  type:         effort_controllers/JointPositionController
  joint:        ur5_shoulder_pan_joint
  pid:          {p: 2000, i: 100, d: 50}

ur5_shoulder_lift_joint_position_controller:
  type:         effort_controllers/JointPositionController
  joint:        ur5_shoulder_lift_joint
  pid:          {p: 2000, i: 100, d: 50}

ur5_elbow_joint_position_controller:
  type:         effort_controllers/JointPositionController
  joint:        ur5_elbow_joint
  pid:          {p: 2000, i: 100, d: 50}

ur5_wrist_1_joint_position_controller:
  type:         effort_controllers/JointPositionController
  joint:        ur5_wrist_1_joint
  pid:          {p: 2000, i: 100, d: 50}

ur5_wrist_2_joint_position_controller:
  type:         effort_controllers/JointPositionController
  joint:        ur5_wrist_2_joint
  pid:          {p: 2000, i: 100, d: 50}

ur5_wrist_3_joint_position_controller:
  type:         effort_controllers/JointPositionController
  joint:        ur5_wrist_3_joint
  pid:          {p: 2000, i: 100, d: 50}

ur5_shoulder_pan_joint_velocity_controller:
  type:         effort_controllers/JointVelocityController
  joint:        ur5_shoulder_pan_joint
  pid:          {p: 100, i: 1, d: 0.1}

ur5_shoulder_lift_joint_velocity_controller:
  type:         effort_controllers/JointVelocityController
  joint:        ur5_shoulder_lift_joint
  pid:          {p: 100, i: 1, d: 0.1}

ur5_elbow_joint_velocity_controller:
  type:         effort_controllers/JointVelocityController
  joint:        ur5_elbow_joint
  pid:          {p: 100, i: 1, d: 0.1}

ur5_wrist_1_joint_velocity_controller:
  type:         effort_controllers/JointVelocityController
  joint:        ur5_wrist_1_joint
  pid:          {p: 100, i: 1, d: 0.1}

ur5_wrist_2_joint_velocity_controller:
  type:         effort_controllers/JointVelocityController
  joint:        ur5_wrist_2_joint
  pid:          {p: 100, i: 1, d: 0.1}

ur5_wrist_3_joint_velocity_controller:
  type:         effort_controllers/JointVelocityController
  joint:        ur5_wrist_3_joint
  pid:          {p: 100, i: 1, d: 0.1}

gripper_joint_effort_controller:
  type: effort_controllers/JointEffortController
  joint: gripper_joint